openrtb-json/src/main/scala/com/powerspace/openrtb/json/util/EncodingUtils.scala

```diff
@@ -69,26 +60,37 @@ object EncodingUtils {
       })
     }
 
-    def cleanRtb: Encoder[T] = renameOneof.clean.transformBooleans
-
-    def rename(renames: Map[String, String]): Encoder[T] =
+    def renameOneof: Encoder[T] =
       encoder.mapJson({
         _.mapObject(obj =>
           JsonObject.fromIterable(obj.toIterable.map {
-            case (key, json) => (renames.getOrElse(key, key), json)
+            case (key, json) => (key.stripSuffix("_oneof"), json)
           }))
       })
 
-    def renameOneof: Encoder[T] =
+    def clean: Encoder[T] = clean(Seq())
+
+    /**
+      * Clean up the JSON file from empty fields/arrays unless present within the given list
+      */
+    def clean(toKeep: Seq[String] = Seq()): Encoder[T] = {
+      encoder.mapJson(jsonObject =>
+        jsonObject.mapObject(obj =>
+          obj.filter {
+            case (_, Json.Null) => false
+            case (id, json) if json.isArray && json.asArray.exists(_.isEmpty) && !toKeep.contains(id) => false
+            case _ => true
+          }))
+    }
+
+    def rename(renames: Map[String, String]): Encoder[T] =
       encoder.mapJson({
         _.mapObject(obj =>
           JsonObject.fromIterable(obj.toIterable.map {
-            case (key, json) => (key.stripSuffix("_oneof"), json)
+            case (key, json) => (renames.getOrElse(key, key), json)
```

Comment:
I don't get this part?

Reply:
I don't get it as well, I'm gonna check it!

Reply:
It's because of scalafmt. `rename` and `renameOneOf` functions have been moved in the file and github doesn't like it. Logic doesn't change